Objects/bytearrayobject.c

```diff
@@ -2163,16 +2470,40 @@ The result is returned as a new bytearray object.
 [clinic start generated code]*/
 
 static PyObject *
-bytearray_join(PyByteArrayObject *self, PyObject *iterable_of_bytes)
-/*[clinic end generated code: output=a8516370bf68ae08 input=aba6b1f9b30fcb8e]*/
+bytearray_join_impl(PyByteArrayObject *self, PyObject *iterable_of_bytes)
+/*[clinic end generated code: output=0ced382b5846a7ee input=49627e07ca31ca26]*/
 {
+    PyObject *ret;
+    Py_BEGIN_CRITICAL_SECTION(self);
```

Comment:
`Py_BEGIN_CRITICAL_SECTION` should be removed as @critical_section` already present